includes/documents/class-wcpdf-invoice.php

```diff
@@ -151,13 +145,35 @@ public function get_sequential_number_store() {
 	}
 	
 	public function get_sequential_number_store_name( $date ) {
-		$year       = $date->date_i18n( 'Y' );
-		$store_name = "{$this->slug}_number_{$year}";
-
-		// legacy filter for backward compatibility
-		$store_name = apply_filters( 'wpo_wcpdf_document_sequential_number_store', $store_name, $this );
+		$year            = $date->date_i18n( 'Y' );
+		$store_base_name = apply_filters( 'wpo_wcpdf_document_sequential_number_store', "{$this->slug}_number", $this );
+	
+		// migrate old stores without year suffix
+		$number_store_migrated = get_option( "wpo_wcpdf_number_store_{$store_base_name}_migrated" );
```

Comment:
rather than working with a per-store option, I would suggest using a global setting to enable or disable migration, which you enable in the plugin upgrade procedure

Reply:
You could make this setting disappear after it's unchecked (and as a result, invisible for new users), and provide an override with a filter.

Reply:
But doing that, if the user doesn't uncheck, the `maybe_migrate` procedure will always try to run right? Do you need that? Because could slow down the `init_number` process.

Reply:
we can remove the procedure after a year, and I don't think `get_option` is much faster than `SHOW TABLES LIKE '{$store_base_name}` but it's something you could measure.